src/ai/susi/server/api/cms/FeedbackLogService.java

```diff
@@ -0,0 +1,145 @@
+
+    @Override
+    public String getAPIPath() {
+        return "/cms/feedbackLog.json";
+    }
+
+    @Override
+    public ServiceResponse serviceImpl(Query call, HttpServletResponse response, Authorization rights, final JsonObjectWithDefault permissions) {
+
+        String model_name = call.get("model", "general");
+        File model = new File(DAO.model_watch_dir, model_name);
+        String group_name = call.get("group", "Knowledge");
+        File group = new File(model, group_name);
+        String language_name = call.get("language", "en");
+        File language = new File(group, language_name);
+        String skill_name = call.get("skill", null);
+        File skill = SusiSkill.getSkillFileInLanguage(language, skill_name, false);
+        String skill_rate = call.get("rating", null);
+        String user_query = call.get("user_query", null);
+        String susi_reply = call.get("susi_reply", null);
+        String country_name = call.get("country_name", null);
+        String country_code = call.get("country_code", null);
+
+
+        JSONObject result = new JSONObject();
+        result.put("accepted", false);
+        if (!skill.exists()) {
+            result.put("message", "skill does not exist");
+            return new ServiceResponse(result);
+
+        }
+        JsonTray skillRating = DAO.feedbackLogs;
+        JSONObject modelName = new JSONObject();
+        JSONObject groupName = new JSONObject();
+        JSONObject languageName = new JSONObject();
+        JSONArray skillName = new JSONArray();
+        JSONObject feedbackLogObject = new JSONObject();
+        Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+        if (skillRating.has(model_name)) {
+            modelName = skillRating.getJSONObject(model_name);
+            if (modelName.has(group_name)) {
+                groupName = modelName.getJSONObject(group_name);
+                if (groupName.has(language_name)) {
+                    languageName = groupName.getJSONObject(language_name);
+                    if (languageName.has(skill_name)) {
+                        skillName = languageName.getJSONArray(skill_name);
+                        feedbackLogObject.put("timestamp", timestamp);
+                        feedbackLogObject.put("feedback", skill_rate);
+                        feedbackLogObject.put("user_query", user_query);
+                        feedbackLogObject.put("susi_reply", susi_reply);
+                        feedbackLogObject.put("country_name", country_name);
+                        feedbackLogObject.put("country_code", country_code);
+                        skillName.put(feedbackLogObject);
+                        languageName.put(skill_name, skillName);
+                        groupName.put(language_name, languageName);
+                        modelName.put(group_name, groupName);
+                        skillRating.put(model_name, modelName, true);
+                        result.put("accepted", true);
```

Comment:
We need the skill path also. Not only the name!

Reply:
done